benchmark/arrayindex.save.js

```diff
@@ -0,0 +1,123 @@
+const Benchmark = require('benchmark')
+const testUtil = require('../test/helpers/utils')
+const _ = require('lodash')
+const async = require('async')
+
+var lounge = require('../')
+var userSchema, User, user, bucket
+
+function connect (options, done) {
+  if (lounge) {
+    lounge.disconnect()
+  }
+
+  lounge = new lounge.Lounge() // recreate it
+
+  var cluster = testUtil.getCluser()
+  bucket = cluster.openBucket('lounge_test', function (err) {
+    if (err) {
+      return done(err)
+    }
+
+    if (options.removeMutateIn) {
+      delete bucket.mutateIn
+      bucket.mutateIn = null
+    }
+
+    lounge.connect({
+      bucket: bucket
+    }, done)
+  })
+}
+
+function setup (options, done) {
+  connect(options, err => {
+    if (err) {
+      return done(err)
+    }
+
+    userSchema = lounge.schema({
+      name: String,
+      email: { type: String, index: true, indexType: 'array' }
+    })
+
+    User = lounge.model('User', userSchema)
+
+    done()
+  })
+}
+
+let counter = 0
+
+function testFnUser (deferred) {
+  counter = counter + 1
+
+  user = new User({
+    name: 'Joe Smith',
+    email: `joe+${counter}@gmail.com`
+  })
+
+  user.save({ waitForIndex: true }, err => {
+    deferred.resolve(err)
+  })
+}
+
+const results = {}
+
+function complete (options, done) {
+  results[options.name] = this.stats.mean * 1000
+  if (options.flush) {
+    bucket.manager().flush(done)
+  } else {
+    process.nextTick(done)
+  }
+}
+
+function createTest (options, testFn) {
+  return function testerFn (done) {
+    console.dir(options, { depth: 3, colors: true })
```

Comment:
💥

Reply:
Nvm, I see they are for benchmarking. Should we be using something else for this?

Reply:
this benchmarking is all very rudimentary and preliminary as I wanted to see what the effects of my changes were. i'm just printing out the options so we can see the progress and know what's benchmarked. `console.dir` is fine in Node.js and just uses `util.inspect()`